scipy/stats/_continuous_distns.py

```diff
@@ -6373,7 +6373,9 @@ def _isf(self, q, c):
         return np.where(q > 0.5, (2.0*(1.0 - q))**(1.0/c), (2*q)**(-1.0/c))
 
     def _munp(self, n, c):
-        return c**2 / (c**2 - n**2)
+        # Return inf if abs(n) >= c
+        with np.errstate(divide='ignore'):
+            return c**2 / np.maximum(c**2 - n**2, 0)
```

Comment:
Test for equality `c == abs(n)` if they are integers and return explicit `np.inf` instead of delegating to numpy for inf handling.

Reply:
Thanks for reviewing. I made a new commit with that change.